### PR TITLE
Add prettier pre-commit hook

### LINKED ISSUE #103: Add prettier as a precommit hook
Now that we have multiple collaborators on the repo, it will be useful to add a prettier pre-commit hook to maintain consistency.

This is how you can setup a pre-commit hook: https://www.npmjs.com/package/prettier#pre-commit-hook

My favorite config is: `prettier --write --single-quote --no-semi`, but I'm open to whatever the majority feels comfortable with 😄 

Thanks to @okonet for [lint-staged](https://github.com/okonet/lint-staged) and @typicode for [husky](https://github.com/typicode/husky)!

### PR DESCRIPTION
**Pull request by hawkins**
Fixes #103 

- Adds Prettier as a simple pre-commit hook via lint-staged
- Ran Prettier on codebase to catch up

### CODE REVIEW

**Comment by SaraVieira** [comment]
Looks great to me! Thanks @hawkins! 🎉🤗

File: package.json [hunk 1]
```diff
@@ -35,6 +36,8 @@
     "github-build": "^1.2.0",
     "glob": "^7.1.2",
     "gzip-size": "^3.0.0",
+    "husky": "^0.14.3",
```
Comment (siddharthkp) [inline package.json]: I think these should be in `devDependendies` instead?

File: package.json [hunk 2]
```diff
@@ -47,5 +50,11 @@
       "path": "./src/files.js",
       "maxSize": "600B"
     }
-  ]
+  ],
+  "lint-staged": {
+    "*.js": [
+      "prettier --write --single-quote --no-semi",
```
Comment (siddharthkp) [inline package.json]: ♥️

File: src/reporter.js [hunk 3]
```diff
@@ -51,7 +51,9 @@ const compare = (files, masterValues = {}) => {
   })
 
   /* prepare the build page */
-  const params = encodeURIComponent(JSON.stringify({ files, repo, branch, commit_message, sha }))
+  const params = encodeURIComponent(
+    JSON.stringify({ files, repo, branch, commit_message, sha })
```
Comment (siddharthkp) [inline src/reporter.js]: 👍

File: package.json (hunk 1 above)
Comment (hawkins) [reply package.json]: You're totally right, good catch - I passed `--dev` instead of `--save-dev`, too used to yarn 😅

Comment (siddharthkp) [reply package.json]: Haha, sorry about that!

I realised some of contributors are using yarn. Which leads to another interesting problem: https://github.com/siddharthkp/bundlesize/issues/106

**Pull request merged**